rpc/src/rpc.rs

```diff
@@ -7355,6 +7514,269 @@ pub mod tests {
+                // 2) Wallet2 - Owns 2 SPL Tokens
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    wallet2_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenOwner).unwrap() == 2);
+                // 3) Mint1 - Is in 2 SPL Accounts
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    mint1_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenMint).unwrap() == 2);
+                // 4) Mint2 - Is in 1 SPL Account
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    mint2_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenMint).unwrap() == 1);
+                // 5) SPL Token Program Owns 6 Accounts
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    inline_spl_token::id(),
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::ProgramId).unwrap() == 6);
+                // 5) System Program Owns 4 Accounts: [Wallet1, Wallet2, SPLTokenProgram]
```

Comment:
This is a bit shaky. Been trying to understand why the system program is coming up with a size of 4. It owns the 2 wallets I created, but what are the other 2 accounts it owns? I'm thinking the SPL Token Program and The SPL Token2022 Program? Need more info.

Reply:
Appears the extra system accounts are the validator startup accounts.

System Program Accounts:

1) 11111113pNDtm61yGF8j2ycAwLEPsuWQXobye5qDR: [lamports:11111111, owner:11111111111111111111111111111111, executable:false, rent_epoch:0]

2) 11111114DhpssPJgSi1YU7hCMfYt1BJ334YgsffXm: [lamports:11111111, owner:11111111111111111111111111111111, executable:false, rent_epoch:0]

3) 8Hzg3d7udASvsDqfGXwMb7g4t3bwLtXMxgnfVAgzC8kN: [lamports:1000000000, owner:11111111111111111111111111111111, executable:false, rent_epoch:0]

4) E6AxyyNZokwFSVLZubUn6iH6mCd9tScNnt4K5JP48acq: [lamports:42, owner:11111111111111111111111111111111, executable:false, rent_epoch:0]